keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/BooleanProxyVertex.java

```diff
@@ -28,12 +29,25 @@ public BooleanProxyVertex(VertexLabel label) {
         this(Tensor.SCALAR_SHAPE, label);
     }
 
+    @ExportVertexToPythonBindings
     public BooleanProxyVertex(long[] shape, VertexLabel label) {
         super(shape);
-        this.setLabel(label);
+        setLabel(label);
+    }
+
+    @Override
+    public void addScopeToLabel(String scope) {
+        super.setLabel(this.getLabel().withExtraNamespace(scope));
+    }
+
+    @Override
+    public <V extends Vertex<BooleanTensor>> V setLabel(VertexLabel label) {
+        if (this.getLabel() != null) {
+            throw new RuntimeException("You should not set the label on a Proxy Vertex");
```

Comment:
I think a more instructive error message is "You should not **change** the label on a Proxy Vertex"